### PR TITLE
41 pm2 prod process

### PR DESCRIPTION
**Pull request by sreynen**
For #41 .

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
-  "presets": ["react", "es2015", "stage-0"],
-  "only": ["*.js", "*.jsx"]
+  "presets": ["react", "es2015", "es2017", "stage-0"],
+  "only": ["*.js", "*.jsx"],
+  "plugins": [
+    "transform-runtime"
```
Comment (shakalee14) [inline .babelrc:5]: great - glad babel-runtime was the way to go

File: README.md [hunk 2]
```diff
@@ -56,7 +56,8 @@ If you need to use Twilio in development but with live keys, do the following to
 
 1. Run `OUTPUT_DIR=./build npm run prod-build-server`
    This will generate something you can deploy to production in ./build and run nodejs server/server/index.js
-2. Run `npm run prod-build-client` ....
+2. Run `npm run prod-build-client`
+3. Make a copy of `spoke-pm2.config.js.template`, e.g. `spoke-pm2.config.js`, add missing creds, and run it with [pm2](https://www.npmjs.com/package/pm2), e.g. `pm2 start spoke-pm2.config.js --env production`
```
Comment (shakalee14) [inline README.md]: instead of creds, can we say 'environment variables'

File: package.json [hunk 3]
```diff
@@ -12,10 +12,10 @@
     "clean": "rm -rf $OUTPUT_DIR",
     "lint": "eslint --fix --ext js --ext jsx src",
     "prod-build-client": "webpack --config ./webpack/config.js",
-    "prod-build-server": "babel ./src -d $OUTPUT_DIR/server --source-maps",
+    "prod-build-server": "babel ./src -d ./build/server --source-maps",
     "prod-build": "npm run clean && npm run prod-build-client && npm run prod-build-server",
     "postinstall": "if [ \"$NODE_ENV\" = production ] ; then npm run prod-build ; fi",
-    "start": "node $OUTPUT_DIR/server/server",
+    "start": "node ./build/server/server",
```
Comment (shakalee14) [inline package.json:19]: do we still need ``OUTPUT_DIR`` as a env variable then if we're going to hardcode ``./build``?

File: spoke-pm2.config.js.template [hunk 4]
```diff
@@ -0,0 +1,140 @@
+const env_production = {
+  NODE_ENV:'production',
+  AWS_ACCESS_KEY_ID:'',
+  AWS_SECRET_ACCESS_KEY:'',
+  APOLLO_OPTICS_KEY:'',
+  DEV_APP_PORT:8090,
+  OUTPUT_DIR:'./build',
```
Comment (shakalee14) [inline spoke-pm2.config.js.template]: I believe we can remove ``OUTPUT_DIR``

File: spoke-pm2.config.js.template [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+   * http://pm2.keymetrics.io/docs/usage/application-declaration/
+   */
+  apps : [
+
+    // First application
+    {
+      name      : 'spoke-server',
+      script    : './build/server/server',
+      env: {
+        COMMON_VARIABLE: 'true'
```
Comment (shakalee14) [inline spoke-pm2.config.js.template]: great! did we ever figure out what ``COMMON_VARIABLE`` does?

**Comment by shakalee14** [comment]
hey @sreynen  did you have a chance to respond to my review?

File: spoke-pm2.config.js.template (hunk 5 above)
Comment (sreynen) [reply spoke-pm2.config.js.template]: I believe that's just example config, not actually used for anything. `env` is just environment variables that get loaded regardless of the `--env` flag, so this is passing an environment variable named `COMMON_VARIABLE` to node, which isn't being used.

File: package.json (hunk 3 above)
Comment (sreynen) [reply package.json:19]: This change was already made to the prod instance before I started working on it, so I don't know what the motivation was.

File: spoke-pm2.config.js.template (hunk 5 above)
Comment (shakalee14) [reply spoke-pm2.config.js.template]: ok - then we can remove it?

File: spoke-pm2.config.js.template [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      script    : './build/server/workers/message-sender-56.js',
+      env_production : env_production
+    },
+    {
+      name      : 'message-sender-789',
+      script    : './build/server/workers/message-sender-789.js',
+      env_production : env_production
+    },
+    {
+      name      : 'number-assigner',
```
Comment (shakalee14) [inline spoke-pm2.config.js.template]: @sreynen shouldn't we remove this number-assigner worker?

**Review by shakalee14 — CHANGES_REQUESTED** [review]
See comment about number-assigner.

**Review by shakalee14 — APPROVED** [review]

**Review by shakalee14 — APPROVED** [review]

**Pull request merged**